src/seeds.accounts.cpp

```diff
@@ -425,10 +426,24 @@ void accounts::update(name user, name type, string nickname, string image, strin
 
 void accounts::send_reward(name beneficiary, asset quantity)
 {
+  // Check balance - if the balance runs out, the rewards run out too.
+  token_accts accts(contracts::token, bankaccts::referrals.value);
+  const auto& acc = accts.get(symbol("SEEDS", 4).code().raw());
+  auto rem_balance = acc.balance;
+  if (quantity > rem_balance) {
+    // Should not fail in case not enough balance
+    // check(false, ("DEBUG: not enough balance on "+bankaccts::referrals.to_string()+" = "+rem_balance.to_string()+", required= "+quantity.to_string()).c_str());
+    return;
+  }
 
-  // TODO: Check balance - if the balance runs out, the rewards run out too.
+  // Checks the current SEEDS price from tlosto.seeds table
+  auto firstprice = pricehistory.rbegin()->seeds_usd;
```

Comment:
this is called "firstprice" but it is actually the last price entry, no?

Reply:
Done, changed it to lastprice